queries/lua/highlights.scm

```diff
@@ -247,3 +247,41 @@
 (string) @string
 
 (escape_sequence) @string.escape
+
+; string.match("123", "%d+")
+(function_call
+  (dot_index_expression
+    field: (identifier) @_method
+    (#any-of? @_method "find" "match"))
+  arguments: (arguments
+               . (_)
+               .
+               (string
+                 content: (string_content) @string.regex)))
+
+(function_call
+  (dot_index_expression
+    field: (identifier) @_method
+    (#any-of? @_method "gmatch" "gsub"))
+  arguments: (arguments
+               . (_)
+               .
+               (string
+                 content: (string_content) @string.regex)))
```

Comment:
These look kinda identical (unless I'm missing something) 🤔

Can't these be combined together?

```scm
(function_call
  (dot_index_expression
    field: (identifier) @_method
    (#any-of? @_method "find" "match" "gmatch" "gsub"))
  (arguments
     . (_)
     . (string (string_content) @string.regex)))
```

Reply:
(and same below)

Reply:
I merged the identical queries in this PR. This feedback I think is also applicable to the `luap` injection counterparts that I copied the queries from. I may address this in a following PR (unless anyone would like it to be done in this one as well)

Reply:
Did you skip the luau queries intentionally?

Reply:
And if you squash these commits, you can add the luap changes as a separate one; no need for a new PR.

Reply:
> Did you skip the luau queries intentionally?

Yes, it turns out that each of the queries was actually slightly different (two using `method_index_expression` and two using `dot_index_expression`, with each query in the different groups using either `(arguments (_) . (string content: _ @string.regex))` or `(arguments (_) (string content: _ @string.regex))`). But I think this is a bug, the `luau`  highlights (and injections) are too liberal (unlike their `lua` counterparts). Example:
![luauhighlightbugwithluap](https://github.com/nvim-treesitter/nvim-treesitter/assets/55766287/7f4dc4de-c5a8-483f-ba5c-6ad90cfeef9b)
The `lua` highlights and injections would not apply to the third argument on line two.

> And if you squash these commits, you can add the luap changes as a separate one; no need for a new PR.

:+1: I will do this to address the above.

Out of curiosity, is there any reason `luau` doesn't just inherit from `lua`?

Reply:
Gotta ask the maintainer.

Reply:
Lua has neovim-specific highlights, obviously luau won't have and won't need them

Reply:
@lucario387 That makes sense, thank you